### PR TITLE
(#8491) - Fix $regex and $ne within $or operator

### PR DESCRIPTION
**Pull request by filionf**

### CODE REVIEW

File: packages/node_modules/pouchdb-selector-core/src/utils.js [hunk 1]
```diff
@@ -308,22 +307,41 @@ function massageSelector(input) {
...
 }
 
+//
+// The $ne and $regex values must be placed in an array because these operators can be used multiple times on the same field.
+// When $and is used, mergeAndedSelectors takes care of putting some of them into arrays, otherwise it's done here.
+//
+function normalizeArrayOperators(selector) {
+  var fields = Object.keys(selector);
+  for (var i = 0; i < fields.length; i++) {
+    var field = fields[i];
```
Comment (garethbowen) [inline packages/node_modules/pouchdb-selector-core/src/utils.js:326]: It'd be better to use a forEach loop here...

```suggestion
  Object.keys(selector).forEach(function (field) {
    ...
  });
```

**Comment by garethbowen** [comment]
Original issue: https://github.com/pouchdb/pouchdb/issues/8491

**Review by garethbowen — CHANGES_REQUESTED** [review]
LGTM. Just one small readability change and I'll sign off on it. Thanks!

File: packages/node_modules/pouchdb-selector-core/src/utils.js (hunk 1 above)
Comment (filionf) [reply packages/node_modules/pouchdb-selector-core/src/utils.js]: Agreed! The forEach is more of my style, I must have been trying to replicate code style from above.

**Review by garethbowen — APPROVED** [review]
Thanks!

**Pull request merged**